packages/calcite-components/src/components/action/action.scss

```diff
@@ -87,146 +104,171 @@
   .text-container--visible {
     @apply w-auto flex-auto opacity-100;
   }
+
+  .action-indicator {
+    @apply w-4 mx-1;
+  }
+}
+
+.indicator-with-icon {
+  @include action-indicator();
 }
 
-:host([data-active]) .button {
+.indicator-without-icon {
+  @apply w-4 mx-1;
+  @include action-indicator();
+}
+
+// Slotted Icon
+::slotted(*:not([name="tooltip"])) {
+  @apply flex;
+}
+:host([loading]) {
+  ::slotted(:not([name="tooltip"])) {
+    @apply hidden;
+  }
+}
+
+:host([data-active]) button {
   @apply focus-inset;
 }
 
+// Scale
 :host([scale="s"]) {
-  .button {
+  button {
     @apply text-n2h px-2 py-1 font-normal;
-  }
-  .button--text-visible .icon-container {
-    margin-inline-end: theme("spacing.2");
+
+    &.button--text-visible .icon-container {
+      margin-inline-end: theme("spacing.2");
+    }
   }
 }
-
 :host([scale="m"]) {
-  .button {
+  button {
     @apply text-n1h px-4 py-3 font-normal;
-  }
-  .button--text-visible .icon-container {
-    margin-inline-end: theme("spacing.3");
+
+    &.button--text-visible .icon-container {
+      margin-inline-end: theme("spacing.3");
+    }
   }
 }
-
 :host([scale="l"]) {
-  .button {
+  button {
     @apply text-0h p-5 font-normal;
-  }
-  .button--text-visible .icon-container {
-    margin-inline-end: theme("spacing.4");
-  }
-}
 
-:host([alignment="center"]) .button {
-  @apply justify-center;
+    &.button--text-visible .icon-container {
+      margin-inline-end: theme("spacing.4");
+    }
+  }
 }
-
-:host([alignment="end"]) .button {
-  @apply justify-end;
+// Compact
+:host([scale="s"][compact]),
+:host([scale="m"][compact]),
+:host([scale="l"][compact]) {
+  button {
```

Comment:
should this go back to .button?

Reply:
no. The styles would need to change if it was applied to anything other than a button anyway. So using a class is just obfuscating what's happening.